compose/service.py

```diff
@@ -953,6 +949,35 @@ def parse_volume_spec(volume_config):
 
     return VolumeSpec(external, internal, mode)
 
+
+def build_volume_from(volume_from_spec):
+    volumes_from = []
+    if isinstance(volume_from_spec.source, Service):
+        containers = volume_from_spec.source.containers(stopped=True)
+        if not containers:
+            volumes_from = ["{}:{}".format(volume_from_spec.source.create_container().id, volume_from_spec.mode)]
+        else:
+            volumes_from = ["{}:{}".format(container.id, volume_from_spec.mode) for container in containers]
+    elif isinstance(volume_from_spec.source, Container):
+        volumes_from = ["{}:{}".format(volume_from_spec.source.id, volume_from_spec.mode)]
+    return volumes_from
```

Comment:
Instead of initializing a list and then overriding it in each branch, can we return immediately from each branch?

```python

if isisntance(...):
   return ...

if isinstance(...):
   return

return []
```